### PR TITLE
Add chat completions client

### PR DESCRIPTION
**Pull request by justin1121**
Plus update some dependencies.

### CODE REVIEW

File: pycape/llms/crypto.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import padding
+from cryptography.hazmat.primitives.ciphers import aead
+
+NONCE_SIZE = 12
+
+
+def aes_decrypt(ctxt: bytes, data_key: bytes) -> bytes:
```
Comment (eric-capeprivacy) [inline pycape/llms/crypto.py]: I would add a test file for this but it's not a huge deal.

File: pycape/llms/llms.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        attestation_doc = await self._ctx.bootstrap(pcrs)
+
+        return attestation_doc
+
+
+class WSMessageType(str, Enum):
+    NONCE = "nonce"
+    ATTESTATION = "attestation"
+    STREAM_CHUNK = "stream_chunk"
+    CHAT_COMPLETIONS_REQUEST = "chat_completion_request"
```
Comment (eric-capeprivacy) [inline pycape/llms/llms.py:154]: How would we version this protocol? I can see the protocol changing like our old ones did.

File: pycape/llms/crypto.py (hunk 1 above)
Comment (justin1121) [reply pycape/llms/crypto.py]: Yeah i can add some tests.

File: pycape/llms/llms.py (hunk 2 above)
Comment (justin1121) [reply pycape/llms/llms.py:154]: We can use the subprotocol parameters when connecting to the server https://stackoverflow.com/questions/67436517/what-is-a-websocket-subprotocol

Comment (justin1121) [reply pycape/llms/llms.py:154]: Which ends up just setting a header.

Comment (eric-capeprivacy) [reply pycape/llms/llms.py:154]: Yeah, we can add that later when the protocol changes

**Review by eric-capeprivacy — APPROVED** [review]

**Pull request merged**